view/frontend/web/recommend.js

```diff
@@ -43,6 +43,9 @@ define([
                             }));
                         },
                         headerComponent({html}) {
+                            if (this.props.recommendations.length === 0) {
```

Comment:
I don't think we should be referencing `this` in the callback as it exposes the underlying implementation. Instead we should destructure the passed arguments to the callback and reference it that way.